charts/ingresslibchart/templates/_ingress.yaml

```diff
@@ -0,0 +1,23 @@
+{{- define "libchart.ingress.tpl" -}}
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: {{ .Chart.Name }}-ingress
+  labels:
+    {{- include "helm.labels" . | nindent 4 }}
+  annotations:
+    networking.gke.io/v1beta1.FrontendConfig: "{{ .Chart.Name }}-ingress-frontendconfig"
+    kubernetes.io/ingress.global-static-ip-name: {{ required "ingress.staticIpName value is required" .Values.ingress.staticIpName | quote }}
+{{- if not (empty .Values.ingress.cert.preSharedCerts) }}
+    ingress.gcp.kubernetes.io/pre-shared-cert: {{ .Values.ingress.cert.preSharedCerts | join "," | quote }}
+{{- end }}
+    kubernetes.io/ingress.allow-http: "false"
+spec:
+  backend:
+    serviceName: {{ .Chart.Name }}-service
+    servicePort: 443
+{{- end -}}
+
+{{- define "libchart.ingress" -}}
+{{- include "libchart.util.merge" (append . "libchart.ingress.tpl") -}}
```

Comment:
I think my helm syntax might be rusty, what is being merged into what here?

Reply:
This merges values that are not included in the library. Here's an [example](https://github.com/broadinstitute/terra-helm/blob/a3fd519477ba6594a5d86acfe0b7e67cf2022282/charts/externalcreds/templates/ingress/ingress.yaml#L2). Here, externalcred as additional values that are not in the library.